### ISSUE TITLE
false-positive(typescript): cannot use `extends` in generic parameter definition

### ISSUE THREAD

**Issue by vegerot** [quick-lint/quick-lint-js#919]
What I'm trying to do: create a function that accepts a generic parameter that must be a subtype of `number`.

```ts
export const getRegistryDetails = async <T extends number = number>(
): Promise<T> => {
  return 1 as T;
}
```

expected: s'all 👌 

actual: 
<img width="845" alt="image" src="https://user-images.githubusercontent.com/5297725/214945014-4cad5237-2686-4688-9cbf-a7cdee175485.png">

```sh
❯ quick-lint-js --language=experimental-typescript hi.ts
hi.ts:1:44: error: unexpected token [E0054]
```

**Comment by vegerot** [comment]
UPDATE: I think the real bug here is #921

`async <T extends number>(){}` is fine.  The problem is `async <T = number>(){}` is busted.

<img width="831" alt="image" src="https://user-images.githubusercontent.com/5297725/214977741-841bfd56-0732-408e-be16-3f79fd3d641e.png">

We can close this issue in favor of #921

**Comment by strager** [comment]
Fixed in commit b5a6409ec7370779ea57b365247fd15f8625053a.
